txtorcon/torconfig.py

```diff
@@ -423,6 +424,11 @@ class EphemeralHiddenService(object):
     https://gitweb.torproject.org/torspec.git/tree/control-spec.txt#n1295
     '''
 
+    @classmethod
+    def _is_valid_keyblob(cls, key_blob_or_type):
+        return (isinstance(key_blob_or_type, str) and
+                re.match(r'[^ :]+:[^ :]+$', key_blob_or_type))
```

Comment:
On the wire, we always send bytes to Tor. But "in general" when interacting with users you get and need to deal with unicode. In this particular case, it should always be 7-bit ASCII so we shouldn't run into problems.

Entire-code-base-wise: yes, more looking at the handling of unicode vs bytes is always good. This wasn't very explicit before 0.19.0 release (i.e. Python3 wasn't supported before then). But, fixing just this one would be good as a first step.

Using the Twisted API directly is fine.